### PR TITLE
fix(pci.objectstorage): add tracking on creation object storage

### PR DESCRIPTION
**Pull request by stif59100**
| Question         | Answer
| ---------------- | ---
 Branch?          | `release/sprint-2242-2244` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-9866
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~
## Description

I added the tracking for creation object storage with tracking page

### CODE REVIEW

**Review by anooparveti — DISMISSED** [review]

**Review by ghyenne — DISMISSED** [review]

File: packages/manager/modules/pci/src/projects/project/storages/containers/add/add.controller.js [hunk 1]
```diff
@@ -190,15 +190,38 @@ export default class PciStoragesContainersAddController {
...
             container: this.container.name,
             userName: this.userModel.createMode?.user?.username,
           }),
-        );
+          trackingTag: `_add::${
+            this.container.offer === 'storage'
+              ? 'standard'
+              : this.container.offer
+          }_${
+            this.container.region?.datacenterLocation
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/containers/add/add.controller.js:207]: `datacenterLocation` could be empty or absent ?
if yes, tracking team except that hit include an `undifined`.

File: packages/manager/modules/pci/src/projects/project/storages/object-storage/object-storage.routing.js [hunk 2]
```diff
@@ -21,7 +24,11 @@ export default /* @ngInject */ ($stateProvider) => {
         ),
     resolve: {
       archive: () => false,
-
+      tagPageParams: /* @ngInject */ (trackPage, $transition$) => {
+        if ($transition$.params().trackingTag) {
+          trackPage($transition$.params().trackingTag);
+        }
+      },
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/object-storage/object-storage.routing.js:31]: i guess we have more sense to complete this into `Ctrl.onInit()` function.
resolve your `tagPageParam` here and send it into your controller.

File: packages/manager/modules/pci/src/projects/project/storages/containers/add/add.controller.js [hunk 3]
```diff
@@ -190,15 +190,38 @@ export default class PciStoragesContainersAddController {
...
+
+        return this.goBackWithTrackingPage({
+          message: this.$translate.instant(message, {
             container: this.container.name,
             userName: this.userModel.createMode?.user?.username,
           }),
-        );
+          trackingTag: `_add::${
+            this.container.offer === 'storage'
+              ? 'standard'
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/storages/containers/add/add.controller.js:204]: Could this test be done once?

**Review by mohammed-zahaf — DISMISSED** [review]

File: packages/manager/modules/pci/src/projects/project/storages/containers/add/add.controller.js [hunk 4]
```diff
@@ -171,11 +171,16 @@ export default class PciStoragesContainersAddController {
   }
 
   add() {
+    const containerOffer =
+      this.container.offer === 'storage' ? 'standard' : this.container.offer;
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/containers/add/add.controller.js:175]: **nitpick:** Please re-use the same variable in Line no.182 as well.

File: packages/manager/modules/pci/src/projects/project/storages/containers/add/add.controller.js [hunk 5]
```diff
@@ -171,11 +171,16 @@ export default class PciStoragesContainersAddController {
...
+    const containerType = this.container.containerType
+      ? `${this.container.containerType}::`
+      : '';
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/containers/add/add.controller.js:178]: **nitpick:** I see that if `containerType` does not exist, it is set to `standard` in the below code. So combine it.
```suggestion
    const containerType = this.container.containerType
      ? `${this.container.containerType}::`
      : 'standard';
```

**Review by anooparveti — APPROVED** [review]

**Review by ghyenne — DISMISSED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Review by ghyenne — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Pull request merged**